src/xdsl/irdl.py

```diff
@@ -431,27 +428,69 @@ def __init__(self, typ: Attribute | type[Attribute] | AttrConstraint):
         super().__init__(typ)
 
 
-def get_variadic_sizes(op: Operation, is_operand: bool) -> list[int]:
+@dataclass(kw_only=True)
+class OpDef:
+    """The internal IRDL definition of an operation."""
+    name: str = field(kw_only=False)
+    operands: list[tuple[str, OperandDef]] = field(default_factory=list)
+    results: list[tuple[str, ResultDef]] = field(default_factory=list)
+    attributes: dict[str, AttributeDef] = field(default_factory=dict)
+    regions: list[tuple[str, RegionDef]] = field(default_factory=list)
+    options: list[IRDLOption] = field(default_factory=list)
+
+    @staticmethod
+    def from_pyrdl(pyrdl_def: type[_OpT]) -> OpDef:
```

Comment:
Who ever came up with pyrdl deserves an award btw.

Reply:
I think this is the idea of Alexander :)